src/meta/processors/admin/AgentHBProcessor.cpp

```diff
@@ -0,0 +1,132 @@
+/* Copyright (c) 2018 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License.
+ */
+
+#include "meta/processors/admin/AgentHBProcessor.h"
+
+#include "common/time/WallClock.h"
+#include "meta/ActiveHostsMan.h"
+#include "meta/KVBasedClusterIdMan.h"
+#include "meta/MetaVersionMan.h"
+
+namespace nebula {
+namespace meta {
+
+AgentHBCounters kAgentHBCounters;
+
+void AgentHBProcessor::onFinished() {
+  if (counters_) {
+    stats::StatsManager::addValue(counters_->numCalls_);
+    stats::StatsManager::addValue(counters_->latency_, this->duration_.elapsedInUSec());
+  }
+
+  Base::onFinished();
+}
+
+// Agent heartbeat register agent to meta and pull all services info in agent's host
+void AgentHBProcessor::process(const cpp2::AgentHBReq& req) {
```

Comment:
Agent heartbeat's interval should be big enough, for example 1min?

Reply:
Agreed.